lib/websocket-endpoint.js

```diff
@@ -147,7 +147,8 @@ class WebsocketEndpoint {
     drain() {
         var self = this;
 
-        if (self._ws.readyState !== WebSocket.OPEN) {
+        if (self._ws.readyState !== WebSocket.OPEN ||
+            self._message_queue.isEmpty()) {
```

Comment:
I agree with your comment that this is the right place to do this check. My understanding is  that it also makes it safe to call `drain()` multiple times back-to-back. Is that true/something we want to support and not have to defend against in other areas?